staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -251,90 +237,95 @@ func (t *tracker) Add(obj runtime.Object) error {
 	if meta.IsListType(obj) {
 		return t.addList(obj, false)
 	}
-
 	objMeta, err := meta.Accessor(obj)
 	if err != nil {
 		return err
 	}
-	return t.add(obj, objMeta.GetNamespace(), false)
-}
-
-func (t *tracker) Create(obj runtime.Object, ns string) error {
-	return t.add(obj, ns, false)
-}
-
-func (t *tracker) Update(obj runtime.Object, ns string) error {
-	return t.add(obj, ns, true)
-}
-
-func (t *tracker) add(obj runtime.Object, ns string, replaceExisting bool) error {
 	gvks, _, err := t.scheme.ObjectKinds(obj)
 	if err != nil {
 		return err
 	}
 	if len(gvks) == 0 {
 		return fmt.Errorf("no registered kinds for %v", obj)
 	}
-
-	t.lock.Lock()
-	defer t.lock.Unlock()
-
 	for _, gvk := range gvks {
-		gr := schema.GroupResource{Group: gvk.Group, Resource: gvk.Kind}
+		gvr, _ := meta.KindToResource(gvk)
```

Comment:
This is the same function used to construct a restmapper, so it's accurate. GVK is only mapped to a single GVR.